core/src/main/java/io/grpc/ClientCallImpl.java

```diff
@@ -85,27 +108,21 @@
     ClientTransport get();
   }
 
+  private enum State {INIT, STARTED, HALF_CLOSED, CANCELED}
```

Comment:
There is a design going on for a state API ... I haven't looked at it yet, but it might be worth taking a peek at.